lua/weapons/weapon_laserpointer.lua

```diff
@@ -0,0 +1,599 @@
+			self:SetOnState(true)
+			self:ButtonSound(true)
+		end
+		local ply = self:GetOwner()
+		local trace = ply:GetEyeTrace()
+		local take = 1 
+		if(self:GetBeamMode())then take = 4 end
+		self:SetClip1(math.max(self:Clip1() - take,0))
+		if(self:GetBeamMode())then LaserPointer_SVBeam(ply,self,ply:EyePos(),ply:GetAimVector()) end
+			timer.Create( self:EntIndex().."LaserPointerOff", 0.3+FrameTime(), 1, function()
+				if(IsValid(self))then
+					self:SetOnState(false)
+					self:ButtonSound(false)
+				end
+			end)
+	self:SetNextPrimaryFire(CurTime()+0.2)
+end
+
+hook.Add( "KeyRelease", "LaserPointerReleaseCheck", function( ply, key )
+    if ( key == IN_ATTACK ) then
+		local pointer = ply:GetActiveWeapon()
+		if(IsValid(pointer) and pointer:GetClass() == "weapon_laserpointer" and pointer:GetOnState() == true)then
+			pointer:SetOnState(false)
+			pointer:SetNextPrimaryFire(CurTime()+0.2)
+			pointer:ButtonSound(false)
+			timer.Destroy( pointer:EntIndex().."LaserPointerOff")
+		end
+    end
+end )
+
+hook.Add("HUDPaintBackground","laserpointer_blinding",function()
+
+end)
+
+function SWEP:OnDrop()
+
+end
+
+local pony_head_hitbones = {}
+pony_head_hitbones["LrigScull"] = true
+pony_head_hitbones["Mane01"] = true
+pony_head_hitbones["Mane02"] = true
+pony_head_hitbones["Mane03"] = true
+pony_head_hitbones["Mane04"] = true
+pony_head_hitbones["Mane05"] = true
+
+local laser_material = Material( "sprites/physg_glow1" )
+local beam_material = Material( "sprites/light_glow02_add" )
+laser_material:SetVector("$color2",Vector(4,4,4))
+beam_material:SetVector("$color2",Vector(4,4,4))
+ 
+
+lpointer_laser_source = lpointer_laser_source or {}
+function LaserPointer_DrawBeam(ply,wep,origin,dir,color,phase,startoverride)
+		if(!CLIENT or !IsValid(ply) or !IsValid(wep) or origin == nil or dir == nil or color == nil)then return end
+		phase = phase or 0
+		if(phase >= 15)then return end
+		local trace = {}
+		trace.start = origin
+		trace.endpos = origin + (dir * 20000) 
+		if(phase == 0)then trace.filter = ply end
+		local bigstart = false
+		local basesize = 8
+		local beammode = wep:GetBeamMode()
+		if(beammode)then basesize = 16 end
+		local tr = util.TraceLine( trace )
+		local beamstart = origin
+		local beamend = tr.HitPos
+		if(tr.Entity == LocalPlayer():GetObserverTarget() or tr.Entity == LocalPlayer())then 
+			local dot = dir:Dot(LocalPlayer():GetAimVector()*-1) 
+			if(tr.Entity:IsPlayer() and math.deg(math.acos(dot)) < 45)then
+				local hitply = tr.Entity
+				--print(hitply:GetHitBoxHitGroup(tr.HitBox,hitply:GetHitboxSet()))
+				--print(hitply:GetBoneName(hitply:GetHitBoxBone(tr.HitBox,hitply:GetHitboxSet())))
+				if(hitply:GetHitBoxHitGroup(tr.HitBox,hitply:GetHitboxSet()) == HITGROUP_HEAD or pony_head_hitbones[hitply:GetBoneName(hitply:GetHitBoxBone(tr.HitBox,hitply:GetHitboxSet()))])then 
+					if(!tr.Entity:ShouldDrawLocalPlayer())then
+						beamend = EyePos() + EyeAngles():Forward()*16
```

Comment:
Instead of using EyeAngles():Forward(), I think (origin-EyePos()):GetNormalized() would make the glare appear where the beam comes from

Reply:
oh yeah, good idea